File: 01-input-output/07-swap-variables/solution.js
import readline from 'readline/promises';
import {stdin as input, stdout as output} from 'process';

const rl = readline.createInterface({
    input, output
});

let firstNumber = await rl.question("Enter first number : ");
let secondNumber = await rl.question("Enter second number : ");
let tempNumber;

firstNumber = parseFloat(firstNumber);
secondNumber = parseFloat(secondNumber);

console.log(`First number: ${firstNumber} | Second number: ${secondNumber}`);

console.log("\nPermutation.....")

tempNumber = firstNumber;
firstNumber = secondNumber;
secondNumber = tempNumber;

console.log(`First number: ${firstNumber} | Second number: ${secondNumber}`);
rl.close()

File: 01-input-output/07-swap-variables/solution.c
#include <stdio.h>

float get_number(const char *label);

int main(void){
    const char *first = "first";
    const char *second = "second";

    float temp_number;

    float first_number = get_number(first);
    float second_number = get_number(second);

    printf("First number: %f | Second number: %f\n", first_number, second_number);
    
    printf("\nPermutation.....\n");

    temp_number = first_number;
    first_number = second_number;
    second_number = temp_number;

    printf("First number: %f | Second number: %f\n", first_number, second_number);
    return 0;
}


float get_number(const char *label){
    printf("Enter %s number : ", label);
    float number;
    scanf("%f", &number);
    return number;
}